test/BoardTests.cpp

```diff
@@ -64,3 +64,32 @@ TEST_F(BoardTests, checkJSONFileReaderExceptions)
                              jfr4.readRows(),
                              jfr4.readCols()), InvalidDimensions);
 }
+TEST_F(BoardTests, checkBoardArgumentBoard_WithGetBoardLinesFunction)
+{
+    std::vector<int> vec = {1,2};
+    std::vector<std::vector<int>> vec2 = {vec,vec};
```

Comment:
Lines and Line types should be used here.

Reply:
done